test/Spender.t.sol

```diff
@@ -0,0 +1,105 @@
+    using SafeERC20 for IERC20;
+    using BalanceSnapshot for Snapshot;
+
+    address user = makeAddr("user");
+    address recipient = makeAddr("recipient");
+    address authorized = makeAddr("authorized");
+    address[] trusted = [authorized];
+
+    AllowanceTarget allowanceTarget;
+    MockERC20 mockERC20 = new MockERC20("MockToken", "MTK", 18);
+    MockDeflationaryERC20 deflationaryERC20 = new MockDeflationaryERC20();
+    MockNoReturnERC20 noReturnERC20 = new MockNoReturnERC20();
+    MockNoRevertERC20 noRevertERC20 = new MockNoRevertERC20();
+    IERC20[] tokens = [IERC20(mockERC20), IERC20(address(deflationaryERC20)), IERC20(address(noReturnERC20)), IERC20(address(noRevertERC20))];
+
+    // effectively a "beforeEach" block
+    function setUp() public {
+        // Deploy
+        allowanceTarget = new AllowanceTarget(trusted);
+
+        // Set user's mock tokens balance and approve
+        setEOABalanceAndApprove(user, address(allowanceTarget), tokens, 100);
+
+        // Label addresses for easier debugging
+        vm.label(user, "User");
+        vm.label(recipient, "Recipient");
+        vm.label(authorized, "Authorized");
+        vm.label(address(allowanceTarget), "AllowanceTarget");
+        vm.label(address(mockERC20), "mockERC20");
+        vm.label(address(deflationaryERC20), "deflationaryERC20");
+        vm.label(address(noReturnERC20), "noReturnERC20");
+        vm.label(address(noRevertERC20), "noRevertERC20");
+    }
+
+    function testCannotSpendFromUserByNotAuthorized() public {
+        vm.expectRevert("AllowanceTarget: not authorized");
+        allowanceTarget.spendFromUserTo(user, address(mockERC20), recipient, 100);
+    }
+
+    function testCannotSpendFromUserInsufficientBalance_NoReturnValueToken() public {
+        uint256 userBalance = noReturnERC20.balanceOf(user);
+        vm.expectRevert("ERC20: transfer amount exceeds balance");
+        vm.prank(authorized);
+        allowanceTarget.spendFromUserTo(user, address(noReturnERC20), recipient, userBalance + 1);
+    }
+
+    function testCannotSpendFromUserInsufficientBalance_ReturnFalseToken() public {
+        uint256 userBalance = noRevertERC20.balanceOf(user);
+        vm.expectRevert("SafeERC20: ERC20 operation did not succeed");
+        vm.prank(authorized);
+        allowanceTarget.spendFromUserTo(user, address(noRevertERC20), recipient, userBalance + 1);
+    }
+
+    function testSpendFromUserTo() public {
+        Snapshot memory fromBalance = BalanceSnapshot.take({ owner: user, token: address(mockERC20) });
+        Snapshot memory toBalance = BalanceSnapshot.take({ owner: recipient, token: address(mockERC20) });
+
+        uint256 amount = 100;
+        vm.prank(authorized);
+        allowanceTarget.spendFromUserTo(user, address(mockERC20), recipient, amount);
+
+        fromBalance.assertChange(-int256(amount));
+        toBalance.assertChange(int256(amount));
+    }
+
+    function testSpendFromUserToWithNoReturnValueToken() public {
```

Comment:
I think we should stick to specific test naming rule. I saw some cases above use underline `...Balance_ReturnFalseToken` but here uses `...WithNoReturnValueToken`. 

I would personally prefer no underline, but both are ok to me, as long as they are in an uniform style.

Reply:
let's drop the underline.